plugins/inputs/haproxy/haproxy_test.go

```diff
@@ -171,59 +174,122 @@ func TestHaproxyDefaultGetFromLocalhost(t *testing.T) {
+	fields["hrsp_other"] = fields["http_response.other"]
+	delete(fields, "http_response.other")
+
+	acc.AssertContainsTaggedFields(t, "haproxy", fields, tags)
+}
+
 func HaproxyGetFieldValues() map[string]interface{} {
 	fields := map[string]interface{}{
-		"active_servers":    uint64(1),
-		"backup_servers":    uint64(0),
-		"bin":               uint64(510913516),
-		"bout":              uint64(2193856571),
-		"check_duration":    uint64(10),
-		"cli_abort":         uint64(73),
-		"ctime":             uint64(2),
-		"downtime":          uint64(0),
-		"dresp":             uint64(0),
-		"econ":              uint64(0),
-		"eresp":             uint64(1),
-		"http_response.1xx": uint64(0),
-		"http_response.2xx": uint64(119534),
-		"http_response.3xx": uint64(48051),
-		"http_response.4xx": uint64(2345),
-		"http_response.5xx": uint64(1056),
-		"lbtot":             uint64(171013),
-		"qcur":              uint64(0),
-		"qmax":              uint64(0),
-		"qtime":             uint64(0),
-		"rate":              uint64(3),
-		"rate_max":          uint64(12),
-		"rtime":             uint64(312),
-		"scur":              uint64(1),
-		"smax":              uint64(32),
-		"slim":              uint64(32),
-		"srv_abort":         uint64(1),
-		"stot":              uint64(171014),
-		"ttime":             uint64(2341),
-		"wredis":            uint64(0),
-		"wretr":             uint64(1),
+		"active_servers":      uint64(1),
+		"backup_servers":      uint64(0),
+		"bin":                 uint64(5228218),
+		"bout":                uint64(303747244),
+		"check_code":          uint64(200),
+		"check_duration":      uint64(3),
+		"check_fall":          uint64(3),
+		"check_health":        uint64(4),
+		"check_rise":          uint64(2),
+		"check_status":        "L7OK",
+		"chkdown":             uint64(84),
+		"chkfail":             uint64(559),
+		"cli_abort":           uint64(690),
+		"ctime":               uint64(1),
+		"downtime":            uint64(3352),
+		"dresp":               uint64(0),
+		"econ":                uint64(0),
+		"eresp":               uint64(21),
+		"http_response.1xx":   uint64(0),
+		"http_response.2xx":   uint64(5668),
+		"http_response.3xx":   uint64(8710),
+		"http_response.4xx":   uint64(140),
+		"http_response.5xx":   uint64(0),
+		"http_response.other": uint64(0),
+		"iid":       uint64(4),
+		"last_chk":  "OK",
+		"lastchg":   uint64(1036557),
+		"lastsess":  int64(1342),
+		"lbtot":     uint64(9481),
+		"mode":      "http",
+		"pid":       uint64(1),
+		"qcur":      uint64(0),
+		"qmax":      uint64(0),
+		"qtime":     uint64(1268),
+		"rate":      uint64(0),
+		"rate_max":  uint64(2),
+		"rtime":     uint64(2908),
+		"sid":       uint64(1),
+		"scur":      uint64(0),
+		"slim":      uint64(2),
+		"smax":      uint64(2),
+		"srv_abort": uint64(0),
+		"status":    "UP",
+		"stot":      uint64(14539),
+		"ttime":     uint64(4500),
+		"weight":    uint64(1),
+		"wredis":    uint64(0),
+		"wretr":     uint64(0),
```

Comment:
Only values changed (to update for new sample data), and new fields added. All the existing fields still kept their names for compatibility.